Source/ProjectFirma.Web/ReportTemplates/ReportTemplateGenerator.cs

```diff
@@ -185,13 +185,23 @@ private static void CorrectImageProblemsAndSaveToDisk(ProjectImage projectImage,
                 }
             }
         }
+
+        /// <summary>Create the report from the template to download, using the template's OriginalCompleteFileName.</summary>
+        public ActionResult GenerateAndDownload() => GenerateAndDownload(ReportTemplate.FileResource.OriginalCompleteFileName);
         
-        public ActionResult GenerateAndDownload()
+        /// <summary>Create the report from the template to download.</summary>
+        public ActionResult GenerateAndDownload(string fileName)

```

Comment:
oh! this change is going to break some of Adrians code that I just merged in. Could you merge develop into your branch and update the new callers of this?